### PR TITLE
Pa02

### PR DESCRIPTION
**Pull request by vsnupoudel**
We have completed writing and testing.

### CODE REVIEW

File: src/pa02/test_pa02.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                           (12, 'ResilientPlayer'), (29, 'LazyPlayer')]
+        assert r.winners_per_type() == {'Player': 2, 'LazyPlayer': 5,
+                                        'ResilientPlayer': 3}
+        assert r.durations_per_type() == {'ResilientPlayer': [21, 13, 12],
+                                          'LazyPlayer': [24, 17, 26, 27, 29],
+                                          'Player': [13, 11]}
+        assert r.players_per_type() == {'Player': 2, 'LazyPlayer': 2,
+                                        'ResilientPlayer': 1}
+
+# Durations per type:  {'ResilientPlayer': [21, 13, 12],
```
Comment (huynhngoc) [inline src/pa02/test_pa02.py]: Not a right way to test these functions but nice try!

File: src/pa02/chutes_simulation.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    and the goal.
+    """
+    default_ladders = ((1, 40), (8, 10), (36, 52), (43, 62), (49, 79),
+                       (65, 82), (68, 85))
+    default_chutes = ((24, 5), (33, 3), (42, 30), (56, 37), (64, 27),
+                      (74, 12), (87, 70))
+
+    def __init__(self,
+                 ladders=default_ladders,
+                 chutes=default_chutes,
```
[the remark this answers is not in the document]
Comment (vsnupoudel) [reply src/pa02/chutes_simulation.py]: Thank you. Pycharm alerted me about this.

File: src/pa02/chutes_simulation.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    def move(self):
+        """Modifies the "move"-method of the super() by adding the given
+        number of drop steps. Drop steps subtracted if the got_ladder variable
+        is True and the number of drop steps is less than dice value.
+        """
+        self.num_moves += 1
+        dice_value = randint(1, 6)
+        self.position += dice_value
+        if self.got_ladder and self.drop_steps <= dice_value:
+            self.position -= self.drop_steps
```
[the remark this answers is not in the document]
Comment (vsnupoudel) [reply src/pa02/chutes_simulation.py]: Thanks.Added the greater than condition also.

File: src/pa02/chutes_simulation.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+            self.players_dict[player.__name__] = \
+                self.player_field.count(player)
+
+        # Fill in winner_per_type_dict with zeros
+        for player in player_field:
+            self.winners_dict[player.__name__] = 0
+
+        # Randomize player field if parameter is set to True
+        if randomize_players:
+            shuffle(self.player_field)
```
[the remark this answers is not in the document]
Comment (vsnupoudel) [reply src/pa02/chutes_simulation.py]: Kept inside the single_game function

File: src/pa02/chutes_simulation.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        pl = []
+        # Creating an object each for each player
+        # Maybe could do this outside the function in __init__
+        for play in self.player_field:
+            pl.append(play())
+
+        while True:
+            for player in pl:
+                player.move()
+                if player.position >= 90:
```
[the remark this answers is not in the document]
Comment (vsnupoudel) [reply src/pa02/chutes_simulation.py]: Using the function now 'if self.board.goal_reached():'

**Pull request merged**